R/github-pat.R

```diff
@@ -1,40 +1,65 @@
 #' Set your Github Personal Access Token
 #'
-#' Populates the `GITHUB_PAT` environment variable using the [git_credential][http_credentials]
-#' manager, which `git` itself uses for storing passwords. The credential manager
-#' returns stored credentials if available, and securely prompt the user for
-#' credentials when needed.
+#' Populates the `GITHUB_PAT` environment variable using a token stored with the
+#' [git_credential][http_credentials] manager, which `git` itself uses for
+#' storing passwords. The credential manager returns stored credentials if
+#' available, and securely prompts the user for credentials when needed.
 #'
 #' Packages that require a `GITHUB_PAT` can call this function to automatically
 #' set the `GITHUB_PAT` when needed. Users may call this function in their
 #' [.Rprofile][Startup] script to automatically set `GITHUB_PAT` for each R
 #' session without hardcoding any tokens on disk in plain-text.
 #'
 #' @export
-#' @param force_new forget existing pat, always ask for new one.
-#' @param validate checks with the github API that this token works. Defaults to
-#' `TRUE` only in an interactive R session (not when running e.g. CMD check).
-#' @param verbose prints a message showing the credential helper and PAT owner.
-#' @return Returns `TRUE` if a valid GITHUB_PAT was set, and FALSE if not.
-set_github_pat <- function(force_new = FALSE, validate = interactive(), verbose = validate){
+#'
+#' @param url A URL that conveys login information, such as protocol (e.g.
+#'   "https"), remote hostname (e.g. "github.com" or
+#'   "github.starkindustries.com"), and username (e.g. "tony"). See the
+#'   input/output documentation for the [`git-credential`
+#'   command](https://git-scm.com/docs/git-credential#IOFMT) for acceptable
+#'   formats.
+#' @param force_new Clear any existing, matching PAT from the store and
+#'   re-prompt the user.
```

Comment:
I reworded this. Somehow the original wording made it sound like you had to enter a new PAT. But it can be an existing PAT, just re-entered.

Reply:
I think my main point is that it needs to be clear that the "forgetting" part of this is purely local, i.e. we're not revoking the actual token.